src/app/service/crew/_components/SortStatus.tsx

```diff
@@ -0,0 +1,39 @@
+import SORT_BY from '@/constants/crew/sortBy';
+import { cn } from '@/lib/utils';
+import { useSortByState } from '@/store/crew/sortsStore';
+import { defaultOptionStyles, selectedOptionStyles } from './UI/commonClass';
+import { useEffect } from 'react';
+
+function SortStatus() {
+  const { selectedOption, setSelectedOption } = useSortByState();
+
+  const clickHandler = (option: (typeof SORT_BY)[keyof typeof SORT_BY]) => {
+    setSelectedOption(option);
+  };
+
+  useEffect(() => {
+    setSelectedOption(SORT_BY.mylist);
+  }, []);
+
+  return (
+    <div className="mt-[16px]">
+      {Object.values(SORT_BY)
+        .slice(4, 6)
+        .map((option, index) => (
+          <button
+            key={index}
+            type="button"
+            onClick={() => clickHandler(option)}
+            className={cn(
+              defaultOptionStyles,
+              selectedOption === option ? selectedOptionStyles : null,
+            )}
+          >
+            {option.title}
+          </button>
```

Comment:
key value를 배열의 Index가 아닌, `uuid`값을 사용하는 것은 어떨까요??

[배열의 인덱스를 key 값으로 사용하지 않는 이유](https://medium.com/sjk5766/react-%EB%B0%B0%EC%97%B4%EC%9D%98-index%EB%A5%BC-key%EB%A1%9C-%EC%93%B0%EB%A9%B4-%EC%95%88%EB%90%98%EB%8A%94-%EC%9D%B4%EC%9C%A0-3ce48b3a18fb)
```suggestion
        .map((option, index) => (
          <button
            key={v4()}
            type="button"
            onClick={() => clickHandler(option)}
            className={cn(
              defaultOptionStyles,
              selectedOption === option ? selectedOptionStyles : null,
            )}
          >
            {option.title}
          </button>
```

Reply:
남겨주신 아티클 덕분에 리액트에서의 배열의 key값 설정에 대해 더 찾아보게 되었는데, 
렌더단에서 uuid를 실행해서 렌더하면 그때마다 새로운 uuid가 할당되기 때문에 "변하지 않은 데이터의 재렌더링을 막는" key의 역할이 제대로 수행되지 않게된다고 이해했는데, 좀 고민을 해보면 좋겠다는 생각이 들었습니다..! 
[React 올바르게 key 사용하기 uuid, index, Math.random() 대체 가능할까?](https://velog.io/@jerrychu/React-%EC%98%AC%EB%B0%94%EB%A5%B4%EA%B2%8C-key-%EC%82%AC%EC%9A%A9%ED%95%98%EA%B8%B0-index-uuid-%EB%8C%80%EC%B2%B4-%EA%B0%80%EB%8A%A5%ED%95%A0%EA%B9%8C)


그리고 프로필사진 배열 등은 api 응답 받을때마다 검증되고 프론트단에서는 수정 및 재배열이 일어나지 않아서 index를 key로 사용해도 무방하다고 판단했습니다. 

>Q2. 그럼 절대로 배열의 index를 key로 사용하면 안되나요?
>
>제가 참고한 [Robin Pokorny ]말에 의하면 아래 3가지를 만족 할 경우 안심하고 index를 key로 써도 된다고 이야기 했습니다.
>
>- 배열과 각 요소가 static이며 computed 되지 않고 변하지 않아야 한다.
>- 데이터 내부에 id로 쓸만한 unique 값이 없을 경우
>- 데이터가 결코 reordered or filtered 되지 않을 경우

(옵션 UI 등)상수로 정의된 배열들도 변경될 일은 없지만 인덱스 대신 option.value 등 유니크한 값들로 변경했습니다! 리뷰 감사합니다 :>

Reply:
오! 저는 깊게 다루지 못했던 부분인데 확인하시고 상세히 피드백해주셔서 감사합니다 :>

덕분에 저도 더 배워가요!